extensions/mssql/src/util/dataCache.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class DataCache {
+
+	millisecondsToLive: number;
+	getValueFunction: (...args: any[]) => any;
+	cache: any;
+	fetchDate: Date;
```

Comment:
Why is this becoming a class member? That just seems weird?

For utility classes like this it helps if you're not relying on some internal state for the cache operations.

Reply:
These items are class members because they need class scope; that is, they are accessed by more than one method.